clients/src/main/java/org/apache/kafka/common/config/ConfigDef.java

```diff
@@ -1256,6 +1256,16 @@ public static class ConfigKey {
         public final boolean internalConfig;
         public final String alternativeString;
 
+        // This constructor is present for backward compatibility reasons.
+        public ConfigKey(String name, Type type, Object defaultValue, Validator validator,
+                         Importance importance, String documentation, String group,
+                         int orderInGroup, Width width, String displayName,
+                         List<String> dependents, Recommender recommender,
+                         boolean internalConfig) {
+            this(name, type, defaultValue, validator, importance, documentation, group, orderInGroup, width, displayName,
+                dependents, recommender, internalConfig, null);
+        }
+
         public ConfigKey(String name, Type type, Object defaultValue, Validator validator,
```

Comment:
As #13909 is not in any release, we can change this constructor from `public` to `private`

Reply:
Done